src/details/ArborX_Ray.hpp

```diff
@@ -198,6 +199,231 @@ bool intersects(Ray const &ray, Box const &box)
   return intersection(ray, box, tmin, tmax) && (tmax >= 0.f);
 }
 
+// The function returns the index of the largest
+// component of the direction vector.
+KOKKOS_INLINE_FUNCTION int findLargestComp(Vector const &dir)
+{
+  int kz = 0;
+  for (int i = 1; i < 3; i++)
+  {
+    float compmax = std::fabs(dir[i - 1]);
+    if (std::fabs(dir[i]) > compmax)
+    {
+      compmax = dir[i];
+      kz = i;
+    }
+  }
+  return kz;
+}
```

Comment:
Consider `dir = {3,1,2}`, your function will return `2` which I expect is not what you wanted is it?

Reply:
Yeah, and also problems with signs. It should be 
```suggestion
KOKKOS_INLINE_FUNCTION int findLargestComp(Vector const &dir)
{
  int kz = 0;
  auto max = std::abs(dir[0]);
  for (int i = 1; i < 3; i++)
  {
    auto f = std::fabs(dir[i]);
    if (std::fabs(f) > max)
    {
      max = f;
      kz = i;
    }
  }
  return kz;
}
```